src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        final Node<T> newNode = new Node<>(last, value, null);
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else {
+            checkIndexForAddition(index);
+            Node<T> nextNode = (index == 0) ? first : findNodeByIndex(index);
+            Node<T> previousNode = nextNode.prev;
+            Node<T> newNode = new Node<>(previousNode, value, nextNode);
+            if (previousNode == null) {
+                first = newNode;
+            } else {
+                previousNode.next = newNode;
+            }
+            nextNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = findNodeByIndex(index);
+        T oldValue = current.item;
+        current.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> nodeToRemove = findNodeByIndex(index);
+        T removedElement = nodeToRemove.item;
+        unlink(nodeToRemove);
+        return removedElement;
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> x = first; x != null; x = x.next) {
+            if (object == x.item || object != null && object.equals(x.item)) {
+                unlink(x);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
```

Comment:
Private methods and classes should be after public ones in your class. Move the `Node` class definition to the end of the `MyLinkedList` class.